### PR TITLE
finish chapter 8

### PR DESCRIPTION
**Pull request by tinhhuu96**
image

### CODE REVIEW

**Comment by tinhhuu96** [comment]
ready

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class SessionsController < ApplicationController
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+    if user && user.authenticate(params[:session][:password])
+      log_in user
+      redirect_to user
+    else
+      flash[:danger] = "Invalid email/password combination"
```
Comment (dothidiemthao) [inline app/controllers/sessions_controller.rb]: dùng i18n

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id]
+  end
+
+  def logged_in?
+    !current_user.nil?
```
Comment (dothidiemthao) [inline app/helpers/sessions_helper.rb]: dùng "current_user.present?" thay cho !current_user.nil?

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  def log_out
+    session.delete(:user_id)
```
Comment (dothidiemthao) [inline app/helpers/sessions_helper.rb]: bỏ ( )

File: app/views/layouts/_header.html.erb [hunk 4]
```diff
@@ -3,9 +3,26 @@
...
+          <li class="dropdown">
+            <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+              Account <b class="caret"></b>
+            </a>
+            <ul class="dropdown-menu">
+              <li><%= link_to "Profile", current_user %></li>
+              <li><%= link_to "Settings", "#" %></li>
+              <li class="divider"></li>
+              <li>
+                <%= link_to "Log out", logout_path, method: :delete %>
```
Comment (dothidiemthao) [inline app/views/layouts/_header.html.erb]: i18n "Log out"

File: app/views/layouts/_header.html.erb [hunk 5]
```diff
@@ -3,9 +3,26 @@
...
+              </li>
+            </ul>
+          </li>
+        <% else %>
+          <li><%= link_to "Log in", login_path %></li>
```
Comment (dothidiemthao) [inline app/views/layouts/_header.html.erb]: "Log in" i18n

**Comment by dothidiemthao** [comment]
commented

**Comment by tinhhuu96** [comment]
ready

**Comment by dothidiemthao** [comment]
merged

**Pull request merged**